roles/pulp_group/tasks/main.yml

```diff
@@ -1,24 +1,48 @@
 ---
 
-- name: Get group list
+- name: Get information for each group
   uri:
-    url: "{{ pulp_group_url }}"
+    url: "{{ pulp_group_url }}?name={{ item.name }}"
     user: "{{ pulp_username }}"
     password: "{{ pulp_password }}"
     method: GET
     status_code: 200
     force_basic_auth: true
-  register: groups_list_result
+  loop: "{{ pulp_groups }}"
+  register: groups_get_result
+
+- name: Reset groups facts
+  set_fact:
+    remove_groups: []
+    create_groups: []
+
+- name: Set group pulp_hrefs to remove fact
+  set_fact:
+    remove_groups: "{{ remove_groups | default([]) + [group.json.results[0]] }}"
+  loop: "{{ groups_get_result.results }}"
+  loop_control:
+    loop_var: "group"
+    label: "{{ group.item.name }}"
+  when:
+   - group.json.count == 1
+   - group.item.state is defined
+   - group.item.state == "absent"
 
-- name: Set fact remove_groups
+- name: Set group names to create fact
   set_fact:
-    remove_groups: "{{ (remove_groups | default([])) + [item.name] }}"
-  when: item.state is defined and item.state == 'absent'
-  with_items: "{{ pulp_groups }}"
+    create_groups: "{{ create_groups | default([]) + [group.item] }}"
+  loop: "{{ groups_get_result.results }}"
+  loop_control:
+    loop_var: "group"
+    label: "{{ group.item.name }}"
+  when:
+   - group.json.count == 0
+   - >
+     group.item.state is not defined
+     or
+     ( group.item.state is defined and group.item.state != "absent" )
```

Comment:
nit: `group.item.state is defined` is redundant

Reply:
Removed, thanks.